src/state/slices/opportunitiesSlice/utils.ts

```diff
@@ -30,5 +30,10 @@ export const filterUserStakingIdByStakingIdCompareFn = (
   return deserializedStakingId === stakingId
 }
 
+// An OpportunityId as an AssetId
 export const toOpportunityId = (...[args]: Parameters<typeof toAssetId>) =>
   toAssetId(args) as OpportunityId
+
+// An OpportunityId as an AccountId
+export const toOpportunityAccountId = (...[args]: Parameters<typeof toAccountId>) =>
```

Comment:
Typing OpportunityIds as AssetIds was technically correct in the EVM world where for all intent and purposes, an ERC20 EOA and a smart contract account are the same (an on-chain address) and also correct for savers where the ID is the L1 AssetId, but now that we're dealing with validators as opportunities, that's a distinction we should make.

Since this lives in a controlled environment where we know that a Cosmos validator is for ATOM, and an Osmosis validator is for OSMO, we will be able to simply `fromAccountId()` the account and check the chain to know which asset the opportunity is for